### PR TITLE
Fix: leave external window open and refresh with updated content

### LINKED ISSUE #48326: Preview external window disappears when moving to the next challenge
### Describe the Issue

I have two screens: I use one to keep the editor open and one to keep the preview window.

When I click on the "Submit and go to the next challenge" button, the preview window disappears: I have to re-open it in a separate window, and re-move it to the second screen.

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-forms-by-building-a-registration-form/step-29

### Steps to Reproduce

1. Open a step of the "(New) Responsive Web Design" curriculum
2. Click on the button to open the preview in a second window (top-right, next to the "Preview" button)
3. Complete the step, click on "Check Your code" and then on "Submit and go to the next challenge"
4. The preview window closes

### Expected behavior

The preview window should stay open, containing the preview of the next step

### Screenshots

_No response_

### System

- Device: Laptop
- OS: Ubuntu 22.04
- Browser: Chrome
- Version: 106

### Additional context

This may be related to https://github.com/freeCodeCamp/freeCodeCamp/issues/47608

### PR DESCRIPTION
**Pull request by KravMaguy**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #48326

Tested locally starting with step # 1 of new responsive web development curriculum and it keeps the external window open and refreshes with content of the next challenge.

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
@KravMaguy fyi, there are some TypeScript errors.

File: client/src/templates/Challenges/components/preview-portal.tsx [hunk 1]
```diff
@@ -51,37 +91,63 @@ class PreviewPortal extends Component<PreviewPortalProps> {
...
+
+    if (portalWindow && showPreviewPortal && portalDocument) {
+      this.existingExternalWindow.document.head.innerHTML = '';
+      this.existingExternalWindow.document.body.innerHTML = '';
+      processWindowAndDocuments(
+        this.titleEl,
+        this.styleEl,
+        this.containerEl,
+        this.existingExternalWindow,
+        this.existingExternalWindow?.document
```
Comment (nicolo-ribaudo) [inline client/src/templates/Challenges/components/preview-portal.tsx:124]: If the document is always `externalWindow.document`, do we still need to store it in the redux store? Isn't storing just the window enough?

Comment (KravMaguy) [reply client/src/templates/Challenges/components/preview-portal.tsx:124]: disreguard this code. It is not good. I will make a new pull request you will see it shortly.

Comment (KravMaguy) [reply client/src/templates/Challenges/components/preview-portal.tsx:124]: @nicolo-ribaudo check the new PR on the issue it is much better. I think you will like it.

**Pull request closed without merging**